internal/parse/parser.go

```diff
@@ -243,58 +236,64 @@ func (p *Parser) skipName() bool {
 //  - bool == false, err == nil
 //		The construct was not the one we are looking for
 
-// parseIdentifier parses either a name made up only of nameBytes or an
-// asterisk.
-func (p *Parser) parseIdentifier() (string, bool) {
-	if p.skipByte('*') {
-		return "*", true
-	}
+type asteriskFlag int
+
+const (
+	allowAsterisk asteriskFlag = iota
+	disallowAsterisk
+)
 
-	idStart := p.pos
+// parseIdentifier parses either a name made up only of nameBytes (or an
+// asterisk if asteriskF=allowAsterisk). On success it returns the parsed string
+// and true. Otherwise, it returns the empty string and false.
+func (p *Parser) parseIdentifier(asteriskF asteriskFlag) (string, bool) {
+	if asteriskF == allowAsterisk {
+		if p.skipByte('*') {
+			return "*", true
+		}
+	}
+	mark := p.pos
 	if p.skipName() {
-		return p.input[idStart:p.pos], true
+		return p.input[mark:p.pos], true
 	}
 	return "", false
 }
 
-// parseFullName parses a column name or Go field name, optionally dot-prefixed by
-// its table or type name respectively.
-func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+// parseGoObject parses a source or target go object of the form Prefix.Name
+// where the type is the Prefix and the field is the Name (this applies to maps
+// and structs). On sccuess it returns the parsed FullName, true and nil.
+// If it cannot parse the Go object it returns false. If the construct was
+// recognised as a Go object there will be an error. Otherwise err=nil.
```

Comment:
I think this comment wasn't very clear, I've reworded it.
`If a Go object is found, but not formatted correctly, false and an error are returned. Otherwise the error is nil.`